### PR TITLE
ARROW-2917: [Python] Use detach() to avoid PyTorch gradient errors

### PR DESCRIPTION
**Pull request by alok**
`detach()` doesn't copy data unless it has to and will give a RuntimeError if the detached data needs to have its gradient calculated.

### CODE REVIEW

**Comment by alok** [comment]
@kou I'd rather not make a JIRA account for a one word change. Would you mind adding the patch? It does pass the tests.

**Comment by wesm** [comment]
Is there any way to trigger the problem that this fixes? Having an ASF JIRA account is a good idea; at some point you may find yourself contributing a larger patch to an Apache project.

**Comment by alok** [comment]
@wesm Here's a script that triggers it for me: https://gist.github.com/c683d79cbc4b5c51282df983d81856ae

**Comment by alok** [comment]
@wesm @kou Created JIRA issue and updated title.

**Comment by wesm** [comment]
I added a unit test; all you had to do was try serializing a tensor with `requires_grad=True`.

**Review by wesm — APPROVED** [review]
+1

**Comment by alok** [comment]
@wesm Thanks for adding the test that I really should've.
